IPython/core/debugger.py

```diff
@@ -324,15 +324,20 @@ def new_do_restart(self, arg):
     def postloop(self):
         self.shell.set_completer_frame(None)
 
-    def print_stack_trace(self):
+    def print_stack_trace(self, context=None):
+        if not context: 
+            context = self.context
+
         try:
             for frame_lineno in self.stack:
-                self.print_stack_entry(frame_lineno, context = 5)
+                self.print_stack_entry(frame_lineno, context=context)
         except KeyboardInterrupt:
             pass
 
     def print_stack_entry(self,frame_lineno,prompt_prefix='\n-> ',
-                          context = 3):
+                          context=None):
+        if not context:
```

Comment:
Side note, here is what 0,1 and 2 do, so that you can judge yourself if it makes sens:
```
ipdb> where 0
  <ipython-input-1-b8ab6ab12ca6>(9)<module>()

> <ipython-input-1-b8ab6ab12ca6>(5)foo()



ipdb> where 1
  <ipython-input-1-b8ab6ab12ca6>(9)<module>()
----> 9 foo()

> <ipython-input-1-b8ab6ab12ca6>(5)foo()
----> 5     raise ValueError('...')



ipdb> where 2
  <ipython-input-1-b8ab6ab12ca6>(9)<module>()
      8     6
----> 9 foo()

> <ipython-input-1-b8ab6ab12ca6>(5)foo()
      4     3
----> 5     raise ValueError('...')
```